### PR TITLE
Migration of ocm tooling from integreatly-operator repo

### PR DESCRIPTION
**Pull request by psturc**
Motivation behind this PR is described in this JIRA:
https://issues.redhat.com/browse/DEL-172

This PR also contains a task from following JIRA (to add this repo to openshift-ci):
https://issues.redhat.com/browse/DEL-178

I'm working on this task as well and the next step is to create a folder for this repo in [openshift/release](https://github.com/openshift/release/tree/master/ci-operator/config/integr8ly) (which is in progress)

### CODE REVIEW

**Comment by psturc** [comment]
ping @davidffrench

File: README.md [hunk 1]
```diff
@@ -175,3 +175,51 @@ $ zcat ~/repos/rhmi-utils/samples/xyz.json.gz | jq -f scripts/resources_by_names
...
+If you want to test your changes on a cluster, the easiest solution would be to spin up OSD 4 cluster using `ocm`. If you want to spin up a cluster using BYOC (your own AWS credentials), follow the additional steps marked as **BYOC**.
+
+#### Prerequisites
+* [OCM CLI](https://github.com/openshift-online/ocm-cli/releases)
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html)
+* [jq](https://stedolan.github.io/jq/)
+
+#### Steps
+
+1. Download the CLI tool and add it to your PATH
```
Comment (davidffrench) [inline README.md]:
```suggestion
1. Download the OCM CLI tool and add it to your PATH
```

File: README.md [hunk 2]
```diff
@@ -175,3 +175,51 @@ $ zcat ~/repos/rhmi-utils/samples/xyz.json.gz | jq -f scripts/resources_by_names
...
+6. Once your cluster is ready, OpenShift Console URL will be printed out together with the `kubeadmin` user & password. These are also saved to `ocm/cluster-credentials.json` file. Also there will be `ocm/cluster.kubeconfig` file created that you can use for running `oc` commands right away, for example, for listing all projects on your OpenShift cluster:
+
+```
+oc --config ocm/cluster.kubeconfig projects
+```
+
+7. If you want to install the latest released RHMI, you can trigger it by applying an RHMI addon.
+Run `make ocm/install/rhmi-addon` to trigger the installation. Once the installation is completed, the installation CR with RHMI components info will be printed to the console.
+
+8. If you want to delete your cluster, run `make ocm/cluster/delete`
```
Comment (davidffrench) [inline README.md]: In the case of BYOC with AWS RDS, Elasticache and S3 resources, does `ocm/cluster/delete` wait for the RHMI CR to be fully removed before deleting the cluster through OCM?

File: README.md [hunk 3]
```diff
@@ -175,3 +175,51 @@ $ zcat ~/repos/rhmi-utils/samples/xyz.json.gz | jq -f scripts/resources_by_names
...
+This command will send a request to [Red Hat OpenShift Cluster Manager](https://cloud.redhat.com/) to spin up your cluster and waits until it's ready. You can see the details of your cluster in `ocm/cluster-details.json` file
+
+6. Once your cluster is ready, OpenShift Console URL will be printed out together with the `kubeadmin` user & password. These are also saved to `ocm/cluster-credentials.json` file. Also there will be `ocm/cluster.kubeconfig` file created that you can use for running `oc` commands right away, for example, for listing all projects on your OpenShift cluster:
+
+```
+oc --config ocm/cluster.kubeconfig projects
+```
+
+7. If you want to install the latest released RHMI, you can trigger it by applying an RHMI addon.
+Run `make ocm/install/rhmi-addon` to trigger the installation. Once the installation is completed, the installation CR with RHMI components info will be printed to the console.
```
Comment (davidffrench) [inline README.md:223]: I don't see a make command with this name `ocm/install/rhmi-addon`

**Review by davidffrench — COMMENTED** [review]
This is great @psturc . Some very small comments from me. I have not tested the commands but I assume they are working as they are copied over from the other repo.

File: README.md (hunk 3 above)
Comment (psturc) [reply README.md:223]: The command is [here](https://github.com/integr8ly/rhmi-utils/pull/20/files#diff-c90e3921bdbf84a8bb9bb3e7179702e5R35-R37)
I talked to Mike about the structure of the scripts, and we agreed that the best solution will be to keep it the same as in integreatly-operator repo, i.e. put all the ocm-related targets into a separate .mk file and import it from the [Makefile in root directory](https://github.com/integr8ly/rhmi-utils/pull/20/files#diff-b67911656ef5d18c4ae36cb6741b7965R1)

File: README.md (hunk 2 above)
Comment (psturc) [reply README.md]: Yes, that's default behaviour. If RHMI CR is found on the cluster, [remove it first and then delete the cluster](https://github.com/integr8ly/rhmi-utils/pull/20/files#diff-e9453c26249c5306c79a77e19f23e49cR135-R139).

**Comment by psturc** [comment]
Thanks for review @davidffrench ! Could you please also add write permissions for this repo to me and @pawelpaszki so we're able to merge things?

File: README.md (hunk 2 above)
Comment (davidffrench) [reply README.md]: Perfect, thanks @psturc

File: README.md (hunk 3 above)
Comment (davidffrench) [reply README.md:223]: Apologies, I missed this. Browser search doesn't work with a `viewed` file in a PR :-)

**Review by davidffrench — APPROVED** [review]
@psturc I have added yourself and @pawelpaszki . You are adding this repo to PROW as well though, correct?

**Comment by psturc** [comment]
Thanks! Yes, I'm currently working on it. I might need to update the base CI image so it contains the `shellcheck` utility to run at least the [lint test](https://github.com/integr8ly/rhmi-utils/pull/20/files#diff-b67911656ef5d18c4ae36cb6741b7965R5) per PR (for the start).
I hope I can finish it during today.

**Comment by psturc** [comment]
^ just minor update to the `.gitignore` & ocm template. Verified the change locally.

**Pull request merged**